internal/cli/login.go

```diff
@@ -12,24 +12,68 @@ import (
 	"github.com/auth0/auth0-cli/internal/prompt"
 )
 
+var (
+	loginAsMachine = Flag{
+		Name:       "Login as machine",
+		LongForm:   "as-machine",
+		Help:       "Initiates authentication as a machine via client credentials (client ID, client secret)",
+		IsRequired: false,
+	}
+	loginTenantDomain = Flag{
+		Name:       "Tenant Domain",
+		LongForm:   "domain",
+		Help:       "Specifies tenant domain when authenticating via client credentials (client ID, client secret)",
+		IsRequired: false,
+	}
+)
+
+type LoginInputs struct {
+	LoginAsMachine bool
+	Domain         string
+	ClientID       string
+	ClientSecret   string
+}
+
 func loginCmd(cli *cli) *cobra.Command {
+	var inputs LoginInputs
+
 	cmd := &cobra.Command{
 		Use:   "login",
 		Args:  cobra.NoArgs,
 		Short: "Authenticate the Auth0 CLI",
-		Long:  "Sign in to your Auth0 account and authorize the CLI to access the Management API.",
+		Long:  "Authenticates the Auth0 CLI either as a user using personal credentials or as a machine using client credentials (client ID/secret).",
+		Example: `
+		auth0 login
+		auth0 login --as-machine
+		auth0 login --as-machine --domain <TENANT_DOMAIN> --client-id <CLIENT_ID> --client-secret <CLIENT_SECRET>
```

Comment:
Should we use lowercase for the placeholders? Like in other commands, e.g. https://github.com/auth0/auth0-cli/blob/main/internal/cli/apps.go#L256